graylog2-web-interface/src/components/event-definitions/event-definition-types/withStreams.jsx

```diff
@@ -25,13 +25,17 @@ export default function withStreams(WrappedComponent, hiddenStreams = []) {
   const wrappedComponentName = WrappedComponent.displayName || WrappedComponent.name || 'Component';
 
   class WithStreams extends React.Component {
-    state = {
-      streams: undefined,
-    };
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        streams: undefined,
+      };
+    }
 
     componentDidMount() {
       StreamsStore.load((streams) => {
-        let filteredStreams = streams;
+        let filteredStreams = streams.filter((s) => s.is_editable);
 
         if (hiddenStreams.length !== 0) {
           filteredStreams = streams.filter((s) => !hiddenStreams.includes(s.id));
```

Comment:
@ousmaneo Regarding the lint error - the only thing modified in that file was removing a constant that was not used anyway and could've caused confusion in the future. I think this PR is complicated enough and no need to add to that with modifying a mostly untouched file.

Reply:
@ousmaneo Sorry, misspoke the first time around on the `hiddenStreams`... too many things in my head 😬

We can remove the `hiddenStreams`, but I had left it there incase anyone has reason to hide a stream in the frontend but not necessarily because the backend said that `is_editable` was false.

If you think it should be fully removed, just let me know. Otherwise, leaves some room for additional functionality should we need it.